ykrt/src/compile/jitc_yk/trace_builder.rs

```diff
@@ -57,7 +57,9 @@ pub(crate) struct TraceBuilder {
     inferred_consts: HashMap<jit_ir::InstIdx, jit_ir::ConstIdx>,
     /// Did this trace end in another frame?
     endframe: TraceEndFrame,
-    /// Whether or not we want blocks to be processed.
+    /// Whether or not we want blocks to be processed. Set to `true` by default and changed to
```

Comment:
Dumb question: why do we need to keep processing blocks at this point? Can't we just immediately stop?

Reply:
We still have to process blocks to count the promotions, otherwise we could stop processing immediately. We could disable that check, but then we might miss it when things with promotions go wrong.

Reply:
Your wish is my command!

Reply:
[4b31ae4](https://github.com/ykjit/yk/pull/1808/commits/4b31ae413bca2a0494ab837ee568247b59f525ea)